### ISSUE TITLE
Incorrect resource url when CKAN is in a subpath

### ISSUE THREAD

**Issue by AdrianMBarrera** [ckan/ckan#2832]
Dear all,

We have a CKAN 2.5.1 instance which is installed in a subpath.
As stated here: http://docs.ckan.org/en/latest/maintaining/configuration.html#ckan-site-url
our ckan.site_url reflects that ( ```ckan.site_url = http://taro-des.fg.ull.es/datos``` ).

We are facing a problem with the url of the resources that happens when SOLR indexes the dataset. For example:

- original url: http://taro-des.fg.ull.es/datos/dataset/0faa5f08-dd15-41f7-a51e-8648174c35de/resource/0c475489-2251-4564-8d4c-2485ac3462d9/download/areas.txt

- indexed url: http://taro-des.fg.ull.es/dataset/0faa5f08-dd15-41f7-a51e-8648174c35de/resource/0c475489-2251-4564-8d4c-2485ac3462d9/download/areas.txt

To us it seems that url_for is not returning the correct url of the resource in resource_dictize function in model_dictize.py, so is causing this problem. 

We have checked the database and there the urls of the resources are stored ok.

Any help would be appreciated. Thanks a lot.

**Comment by wardi** [comment]
The docs seem to be wrong here :-( Your subpath should be specified in http://docs.ckan.org/en/latest/maintaining/configuration.html#ckan-root-path not site_url

**Comment by AdrianMBarrera** [comment]
Thank you @wardi 

We have set the ckan.root_path setting but it results in: 
```TypeError: object of type 'Locale' has no len()```

See the full stack-trace below:

```
[Thu Jan 14 09:30:41.288960 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713] mod_wsgi (pid=14659): Exception occurred processing WSGI script '/etc/ckan/default/apache.wsgi'.
[Thu Jan 14 09:30:41.289020 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713] Traceback (most recent call last):
[Thu Jan 14 09:30:41.289057 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/paste/cascade.py", line 130, in __call__
[Thu Jan 14 09:30:41.289111 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.apps[-1](environ, start_response)
[Thu Jan 14 09:30:41.289137 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/config/middleware.py", line 228, in __call__
[Thu Jan 14 09:30:41.289175 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.app(environ, start_response)
[Thu Jan 14 09:30:41.289198 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/paste/registry.py", line 379, in __call__
[Thu Jan 14 09:30:41.289231 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     app_iter = self.application(environ, start_response)
[Thu Jan 14 09:30:41.289254 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/repoze/who/middleware.py", line 87, in __call__
[Thu Jan 14 09:30:41.289289 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     app_iter = app(environ, wrapper.wrap_start_response)
[Thu Jan 14 09:30:41.289326 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/middleware.py", line 214, in __call__
[Thu Jan 14 09:30:41.289363 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     self.app, new_environ, catch_exc_info=True)
[Thu Jan 14 09:30:41.289385 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/util.py", line 94, in call_wsgi_application
[Thu Jan 14 09:30:41.289417 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     app_iter = application(environ, start_response)
[Thu Jan 14 09:30:41.289439 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/weberror/errormiddleware.py", line 165, in __call__
[Thu Jan 14 09:30:41.289471 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.application(environ, start_response)
[Thu Jan 14 09:30:41.289492 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/dec.py", line 147, in __call__
[Thu Jan 14 09:30:41.289524 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     resp = self.call_func(req, *args, **self.kwargs)
[Thu Jan 14 09:30:41.289545 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/dec.py", line 208, in call_func
[Thu Jan 14 09:30:41.289576 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.func(req, *args, **kwargs)
[Thu Jan 14 09:30:41.289597 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/fanstatic/publisher.py", line 234, in __call__
[Thu Jan 14 09:30:41.289629 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return request.get_response(self.app)
[Thu Jan 14 09:30:41.289650 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/request.py", line 1053, in get_response
[Thu Jan 14 09:30:41.289681 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     application, catch_exc_info=False)
[Thu Jan 14 09:30:41.289702 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/request.py", line 1022, in call_application
[Thu Jan 14 09:30:41.289732 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     app_iter = application(self.environ, start_response)
[Thu Jan 14 09:30:41.289753 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/dec.py", line 147, in __call__
[Thu Jan 14 09:30:41.289784 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     resp = self.call_func(req, *args, **self.kwargs)
[Thu Jan 14 09:30:41.289805 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/dec.py", line 208, in call_func
[Thu Jan 14 09:30:41.289835 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.func(req, *args, **kwargs)
[Thu Jan 14 09:30:41.289856 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/fanstatic/injector.py", line 54, in __call__
[Thu Jan 14 09:30:41.289886 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     response = request.get_response(self.app)
[Thu Jan 14 09:30:41.289907 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/request.py", line 1053, in get_response
[Thu Jan 14 09:30:41.289949 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     application, catch_exc_info=False)
[Thu Jan 14 09:30:41.289972 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/webob/request.py", line 1022, in call_application
[Thu Jan 14 09:30:41.290002 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     app_iter = application(self.environ, start_response)
[Thu Jan 14 09:30:41.290024 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/config/middleware.py", line 389, in inner
[Thu Jan 14 09:30:41.290056 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     result = application(environ, start_response)
[Thu Jan 14 09:30:41.290077 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/beaker/middleware.py", line 73, in __call__
[Thu Jan 14 09:30:41.290108 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.app(environ, start_response)
[Thu Jan 14 09:30:41.290129 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/beaker/middleware.py", line 155, in __call__
[Thu Jan 14 09:30:41.290160 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.wrap_app(environ, session_start_response)
[Thu Jan 14 09:30:41.290181 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/routes/middleware.py", line 131, in __call__
[Thu Jan 14 09:30:41.290212 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     response = self.app(environ, start_response)
[Thu Jan 14 09:30:41.290232 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/wsgiapp.py", line 125, in __call__
[Thu Jan 14 09:30:41.290263 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     response = self.dispatch(controller, environ, start_response)
[Thu Jan 14 09:30:41.290284 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/wsgiapp.py", line 324, in dispatch
[Thu Jan 14 09:30:41.290314 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return controller(environ, start_response)
[Thu Jan 14 09:30:41.290335 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 337, in __call__
[Thu Jan 14 09:30:41.290368 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     res = WSGIController.__call__(self, environ, start_response)
[Thu Jan 14 09:30:41.290390 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/controllers/core.py", line 221, in __call__
[Thu Jan 14 09:30:41.290421 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     response = self._dispatch_call()
[Thu Jan 14 09:30:41.290443 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/controllers/core.py", line 172, in _dispatch_call
[Thu Jan 14 09:30:41.290472 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     response = self._inspect_call(func)
[Thu Jan 14 09:30:41.290493 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/controllers/core.py", line 107, in _inspect_call
[Thu Jan 14 09:30:41.290523 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     result = self._perform_call(func, args)
[Thu Jan 14 09:30:41.290553 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/controllers/core.py", line 60, in _perform_call
[Thu Jan 14 09:30:41.290586 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return func(**args)
[Thu Jan 14 09:30:41.290607 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/controllers/error.py", line 46, in document
[Thu Jan 14 09:30:41.290639 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return render('error_document_template.html')
[Thu Jan 14 09:30:41.290660 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 216, in render
[Thu Jan 14 09:30:41.290692 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     loader_class=loader_class)
[Thu Jan 14 09:30:41.290713 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/templating.py", line 249, in cached_template
[Thu Jan 14 09:30:41.290743 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return render_func()
[Thu Jan 14 09:30:41.290763 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 154, in render_template
[Thu Jan 14 09:30:41.290793 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return render_jinja2(template_name, globs)
[Thu Jan 14 09:30:41.290814 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 104, in render_jinja2
[Thu Jan 14 09:30:41.290843 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return template.render(**extra_vars)
[Thu Jan 14 09:30:41.290864 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/jinja2/environment.py", line 894, in render
[Thu Jan 14 09:30:41.290895 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.environment.handle_exception(exc_info, True)
[Thu Jan 14 09:30:41.290916 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/error_document_template.html", line 1, in top-level template code
[Thu Jan 14 09:30:41.290948 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% extends "page.html" %}
[Thu Jan 14 09:30:41.290968 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/page.html", line 1, in top-level template code
[Thu Jan 14 09:30:41.290998 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% extends "base.html" %}
[Thu Jan 14 09:30:41.291019 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/ckanext/geoview/templates/base.html", line 1, in top-level template code
[Thu Jan 14 09:30:41.291051 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% ckan_extends %}
[Thu Jan 14 09:30:41.291072 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/base.html", line 103, in top-level template code
[Thu Jan 14 09:30:41.291102 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {%- block page %}{% endblock -%}
[Thu Jan 14 09:30:41.291123 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/page.html", line 126, in block "page"
[Thu Jan 14 09:30:41.291152 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {%- block footer %}
[Thu Jan 14 09:30:41.291173 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/page.html", line 127, in block "footer"
[Thu Jan 14 09:30:41.291214 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% include "footer.html" %}
[Thu Jan 14 09:30:41.291236 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/footer.html", line 3, in top-level template code
[Thu Jan 14 09:30:41.291267 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% block footer_content %}
[Thu Jan 14 09:30:41.291288 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/footer.html", line 26, in block "footer_content"
[Thu Jan 14 09:30:41.291317 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% block footer_lang %}
[Thu Jan 14 09:30:41.291338 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/footer.html", line 27, in block "footer_lang"
[Thu Jan 14 09:30:41.291369 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     {% snippet "snippets/language_selector.html" %}
[Thu Jan 14 09:30:41.291390 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/jinja_extensions.py", line 255, in _call
[Thu Jan 14 09:30:41.291422 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return base.render_snippet(args[0], **kwargs)
[Thu Jan 14 09:30:41.291443 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 79, in render_snippet
[Thu Jan 14 09:30:41.291474 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     renderer='snippet')
[Thu Jan 14 09:30:41.291495 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 216, in render
[Thu Jan 14 09:30:41.291526 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     loader_class=loader_class)
[Thu Jan 14 09:30:41.291547 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/templating.py", line 249, in cached_template
[Thu Jan 14 09:30:41.291576 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return render_func()
[Thu Jan 14 09:30:41.291597 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 154, in render_template
[Thu Jan 14 09:30:41.291628 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return render_jinja2(template_name, globs)
[Thu Jan 14 09:30:41.291649 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/base.py", line 104, in render_jinja2
[Thu Jan 14 09:30:41.291679 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return template.render(**extra_vars)
[Thu Jan 14 09:30:41.291700 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/jinja2/environment.py", line 894, in render
[Thu Jan 14 09:30:41.291731 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return self.environment.handle_exception(exc_info, True)
[Thu Jan 14 09:30:41.291753 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/templates/snippets/language_selector.html", line 7, in top-level template code
[Thu Jan 14 09:30:41.291785 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     <option value="{% url_for current_url, locale=locale %}" {% if locale == current_lang %}selected="selected"{% endif %}>
[Thu Jan 14 09:30:41.291807 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/jinja_extensions.py", line 284, in _call
[Thu Jan 14 09:30:41.291850 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return h.url_for(*args, **kwargs)
[Thu Jan 14 09:30:41.291873 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/helpers.py", line 205, in url_for
[Thu Jan 14 09:30:41.291904 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     url = _add_i18n_to_url(my_url, locale=locale, **kw)
[Thu Jan 14 09:30:41.291925 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/ckan/default/src/ckan/ckan/lib/helpers.py", line 304, in _add_i18n_to_url
[Thu Jan 14 09:30:41.291956 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     root_path = re.sub('{{LANG}}', locale, root_path)
[Thu Jan 14 09:30:41.291978 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/python2.7/re.py", line 151, in sub
[Thu Jan 14 09:30:41.292008 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     return _compile(pattern, flags).sub(repl, string, count)
[Thu Jan 14 09:30:41.292029 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/python2.7/re.py", line 273, in _subx
[Thu Jan 14 09:30:41.292059 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     template = _compile_repl(template, pattern)
[Thu Jan 14 09:30:41.292080 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/python2.7/re.py", line 258, in _compile_repl
[Thu Jan 14 09:30:41.292110 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     p = sre_parse.parse_template(repl, pattern)
[Thu Jan 14 09:30:41.292131 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/python2.7/sre_parse.py", line 706, in parse_template
[Thu Jan 14 09:30:41.292160 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     s = Tokenizer(source)
[Thu Jan 14 09:30:41.292181 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/python2.7/sre_parse.py", line 181, in __init__
[Thu Jan 14 09:30:41.292211 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     self.__next()
[Thu Jan 14 09:30:41.292231 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]   File "/usr/lib/python2.7/sre_parse.py", line 183, in __next
[Thu Jan 14 09:30:41.292261 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713]     if self.index >= len(self.string):
[Thu Jan 14 09:30:41.292295 2016] [:error] [pid 14659:tid 140120053233408] [client 127.0.0.1:38713] TypeError: object of type 'Locale' has no len()
```

If it helps, these are our locale settings:

```
## Internationalisation Settings
ckan.locale_default = es
ckan.locale_order = es en pt_BR ja it cs_CZ ca fr el sv sr sr@latin no sk fi ru de pl nl bg ko_KR hu sa sl lv
ckan.locales_offered = es
ckan.locales_filtered_out = en_GB
ckan.root_path = /datos/{{LANG}}
```

Did we miss something?

**Comment by wardi** [comment]
previous discussion on site_url vs. root_path #2599  This bug might be related to the url_for one: it's expecting a sting but getting a Locale object instead

**Comment by wardi** [comment]
fixed by #2851
